snapd-qt/snap.cpp

```diff
@@ -330,6 +364,23 @@ QSnapdEnums::SnapType QSnapdSnap::snapType () const
     }
 }
 
+QStringList QSnapdSnap::source_code () const
+{
+    GPtrArray *source_code = snapd_snap_get_source_code (SNAPD_SNAP (wrapped_object));
+    QStringList result;
+    if (source_code != nullptr) {
+        for (guint i = 0; i < source_code->len; ++i) {
+            // Access the element using g_ptr_array_index and append to QStringList
+            char* source_code_entry = static_cast<char*>(g_ptr_array_index(source_code, i));
+            result.append(QString::fromUtf8(source_code_entry));
+        }
+        // Free the GPtrArray after usage
+        g_ptr_array_free(source_code, TRUE);
```

Comment:
This shouldn't be freed - it's not copied when returned from `snapd_snap_get_source_code`